src/axom/quest/MarchingCubesAlgo.cpp

```diff
@@ -0,0 +1,614 @@
+                  "You must call set_output_mesh before compute_iso_surface.");
+  SLIC_ASSERT_MSG(
+    !_fcnPath.empty(),
+    "You must call set_function_field before compute_iso_surface.");
+
+  /*
+    Notes: how to handle non-interleaved coordinates.
+    We should be able to handle any blueprint-compatible domains.
+
+    Need an array interface that supports strides.
+
+    conduit DataArray uses DataType, which has strides.  But I'm not convinced
+    that it actually uses those strides to compute offsets.  Need to do a practice
+    problem or try it here.
+
+    axom::ArrayView doesn't have strides.  It can support multicomponent arrays
+    so I can represent an Nx3 or 3xN layout.  However, I would have to switch
+    point index and dimension index depending on which layout.  Won't work.
+  */
+
+  const conduit::Node& coordValues =
+    _dom->fetch_existing(_coordsetPath + "/values");
+  bool isInterleaved = conduit::blueprint::mcarray::is_interleaved(coordValues);
+  const int coordSp = isInterleaved ? _ndim : 1;
+  const double* xPtr = coordValues["x"].as_double_ptr();
+  const double* yPtr = _ndim >= 2 ? coordValues["y"].as_double_ptr() : nullptr;
+  const double* zPtr = _ndim >= 3 ? coordValues["z"].as_double_ptr() : nullptr;
+
+  auto& fcnValues = _dom->fetch_existing(_fcnPath + "/values");
+  const double* fcnPtr = fcnValues.as_double_ptr();
+
+  const int* maskPtr = nullptr;
+  if(!_maskPath.empty())
+  {
+    auto& maskValues = _dom->fetch_existing(_maskPath + "/values");
+    maskPtr = maskValues.as_int_ptr();
+  }
+
+  if(!_cellIdField.empty() &&
+     !_surfaceMesh->hasField(_cellIdField, axom::mint::CELL_CENTERED))
+  {
+    _surfaceMesh->createField<int>(_cellIdField, axom::mint::CELL_CENTERED);
+  }
+
+  _contourVal = contourVal;
+
+  if(_ndim == 2)
+  {
+    /*
+      For now, assume zero offsets and zero ghost width.
+      Eventually, we'll have to support index offsets to
+      handle data with ghosts.
+
+      By using ArrayView, we're assuming row-major layout.  To support
+      column-major layout as well, we have to extend ArrayView to
+      compute column-major strides.  We should also use an iteration
+      scheme that's efficient for both layouts.
+    */
+    const axom::StackArray<axom::IndexType, 2> cShape {_cShape[0], _cShape[1]};
+    const axom::StackArray<axom::IndexType, 2> pShape {1 + _cShape[0],
+                                                       1 + _cShape[1]};
+    axom::ArrayView<const double, 2> xView(xPtr, pShape, coordSp);
+    axom::ArrayView<const double, 2> yView(yPtr, pShape, coordSp);
+    axom::ArrayView<const double, 2> fcnView(fcnPtr, pShape);
+    axom::ArrayView<const int, 2> maskView(maskPtr, cShape);
+
+    // Write as regular nested loops.
+    for(int i = 0; i < cShape[0]; ++i)
+    {
+      for(int j = 0; j < cShape[1]; ++j)
+      {
+        const bool skipZone = maskPtr && bool(maskView(i, j));
+        if(!skipZone)
+        {
+          double vfs[4];
```

Comment:
what does `vfs` stand for?

Reply:
Not sure.  :}  That was what the app was using.  I'll change it to `nodalValues`.